pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -159,18 +216,20 @@ func (a *AmbientIndex) updateWaypoint(sa model.WaypointScope, addr *workloadapi.
 	return updates
 }
 
-// All return all known workloads. Result is un-ordered
+// All return all known workloads and services. Result is un-ordered
 func (a *AmbientIndex) All() []*model.AddressInfo {
 	a.mu.RLock()
 	defer a.mu.RUnlock()
 	res := make([]*model.AddressInfo, 0, len(a.byPod)+len(a.serviceByAddr)+len(a.byWorkloadEntry))
+	for _, services := range a.serviceByHostname {
```

Comment:
NOTE: this was needed to support headless services, which don't have an address.